src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -104,13 +107,16 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                 var contextForProject = CreateRemoteWalkContext(_request, _logger);
 
                 CacheFile cacheFile = null;
+                DependencyGraphSpec dgSpec = null;
                 using (var noOpTelemetry = TelemetryActivity.CreateTelemetryActivityWithNewOperationIdAndEvent(parentId: _operationId, eventName: RestoreNoOpInformation))
                 {
                     if (NoOpRestoreUtilities.IsNoOpSupported(_request))
                     {
                         noOpTelemetry.StartIntervalMeasure();
 
-                        var cacheFileAndStatus = EvaluateCacheFile();
+                        dgSpec = NoOpRestoreUtilities.GetDGSpec(_request);
```

Comment:
Let me try to make it clear. 

The only change here is to take dgSpec generation `NoOpRestoreUtilities.GetDGSpec(_request)` out of `EvaluateCacheFile()` api and instead pass it in so that we can use `dgSpec` at later point of time as well. Now I don't follow what is the confusion here?

Reply:
I get that....but that method does additional things.
 
You already have the dgspec on the request. 
What this method does is strips things that specific to noop restore. 
No one else should be concerned with that. 
The reason why that method clones the dg spec is to avoid impacting the dg spec for future operations. 

We can discuss offline.

Reply:
ahh got it... if that's only specific to NoOp then yes I can revert this.